interactive_engine/groot-server/src/main/java/com/alibaba/graphscope/groot/servers/ir/GrootMetaFetcher.java

```diff
@@ -42,13 +47,14 @@ public Optional<IrMeta> fetch() {
             Long snapshotId = entry.getKey();
             GraphSchema schema = entry.getValue();
 
-            try {
-                // TODO: add statistics, otherwise, the CBO will not work
-                return Optional.of(
-                        new IrMeta(
-                                new SnapshotId(true, snapshotId), new IrGraphSchema(schema, true)));
-            } catch (IOException e) {
-                throw new RuntimeException(e);
+            SnapshotId si = new SnapshotId(true, snapshotId);
+            IrGraphSchema irSchema = new IrGraphSchema(schema, true);
+            GraphStatistics statistics = this.schemaFetcher.getStatistics();
```

Comment:
It's not okay to place getting statistics with schema for now. For larger graph, getting statistics is a very time-costly operation, and it is operated at the frequency as schema? I don't think so.

Reply:
We have separated the two interfaces for getting schema (+procedures) and statistics.